### ISSUE TITLE
JS apprentices likely to need some guidance around how to write OO code in JS

### ISSUE THREAD

**Issue by mrmanc** [MCR-Digital/apprentice-boot-camp-fundamentals-1#78]
We should create some slides which describe an approach to doing this, and possibly give them some example code in the repo that they can refer to.

**Comment by helloheatherw** [comment]
Is this specifically related to the exercises? If so, I think it might be worth refactoring the existing starting code as it currently mixes up some new and old syntax, as well as using some non-idiomatic JS that I think caused some confusion.

For example, the starting code for the cards class uses the Class keyword, but also uses prototype syntax for declaring methods. It would also be useful to have this example use the constructor method.

I'm happy to put some thoughts on to PR if that would be helpful.

**Comment by mrmanc** [comment]
> Is this specifically related to the exercises?

No, I was thinking general case stuff to illustrate the syntax. I was thinking it would be good to show both approaches (but unmixed).

> it might be worth refactoring the existing starting code

Sounds sensible. I’m not qualified to say what approach it should follow, but I like the idea of creating a PR and running it past some of the people who support other JS apprentices.
